From 6664d7e7fccf5b17e16ee302c7eb171a5385a337 Mon Sep 17 00:00:00 2001
From: Suresh Gupta <suresh.gupta@xilinx.com>
Date: Fri, 5 Oct 2018 19:43:05 +0530
Subject: [PATCH 0458/1852] v4l: xilinx-multi-scaler: Add number of frames
 processed per channel

commit 61b32c53bb3400688f14ebe072e36953723715c6 from
https://github.com/Xilinx/linux-xlnx.git

Patch helps in debugging to indicate the number of frames
processed by the channel.

Signed-off-by: Suresh Gupta <suresh.gupta@xilinx.com>
Reviewed-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-multi-scaler.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/media/platform/xilinx/xilinx-multi-scaler.c b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
index 9e5be6366bb7..c7ff99f76747 100644
--- a/drivers/media/platform/xilinx/xilinx-multi-scaler.c
+++ b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
@@ -268,6 +268,7 @@ struct xm2msc_q_data {
  * @minor: Minor number of the video device
  * @status: channel status, CHAN_ATTACHED or CHAN_OPENED
  * @taps: number of hwtaps required for channel
+ * @frames: number of frames processed
  * @vfd: V4L2 device
  * @fh: v4l2 file handle
  * @m2m_dev: m2m device
@@ -281,6 +282,7 @@ struct xm2msc_chan_ctx {
 	u32 minor;
 	u8 status;
 	u32 taps;
+	unsigned long frames;
 
 	struct video_device vfd;
 	struct v4l2_fh fh;
@@ -694,6 +696,7 @@ xm2msc_pr_allchanreg(struct xm2m_msc_dev *xm2msc)
 		dev_dbg(dev, "Regs val for channel %d\n", i);
 		dev_dbg(dev, "______________________________________________\n");
 		xm2msc_pr_chanreg(dev, chan_ctx->regs);
+		dev_dbg(dev, "processed frames = %lu\n", chan_ctx->frames);
 		dev_dbg(dev, "______________________________________________\n");
 	}
 }
@@ -955,6 +958,7 @@ static void xm2msc_job_done(struct xm2m_msc_dev *xm2msc)
 			v4l2_m2m_buf_done(dst_vb, VB2_BUF_STATE_DONE);
 			spin_unlock_irqrestore(&xm2msc->lock, flags);
 		}
+		chan_ctx->frames++;
 	}
 }
 
@@ -1636,6 +1640,7 @@ static int xm2msc_open(struct file *file)
 	chan_ctx->fh.m2m_ctx = chan_ctx->m2m_ctx;
 	chan_ctx->status |= CHAN_OPENED;
 	chan_ctx->xm2msc_dev = xm2msc;
+	chan_ctx->frames = 0;
 	xm2msc_set_chan(chan_ctx, true);
 
 	v4l2_info(&xm2msc->v4l2_dev, "Channel %d instance created\n", chan);
-- 
2.31.1

